apps/dashboard/src/components/editor/use-editor-content.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from '../../data';
+import UnpublishedChangesNotice from './unpublished-changes-notice';
+import { EDITOR_VIEW_DRAFT } from '../../util/editor';
+
+// Do something about the timeout - if no project ID should be 500 ?
+export const useEditorContent = ( projectId, editorView ) => {
+	const [ ready, setReady ] = useState( ! projectId );
+
+	const { removeNotice } = useDispatch( 'core/notices' );
+	const {
+		updateEditorContent,
+		restoreEditorContent,
+		setEditorContentChanged,
+	} = useDispatch( STORE_NAME );
+
+	// If projectId or editorView changes, reset 'ready'
+	useEffect( () => {
+		setReady( false );
+	}, [ projectId, editorView ] );
+
+	return useCallback(
+		// using debounce here so we don't pound the state synchronization
+		debounce( ( content ) => {
```

Comment:
Pound it! The whole idea of moving the synchronization into the state layer is that the state layer is always up to date.  
If anything we can just save `content` to the editor state as is and `parse()` it before we push it to `saveAndUpdateProject`, so we don't have to do that part on every keystroke.